geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -1094,6 +1096,13 @@ protected void initializeAction(EnumListenerEvent operation) {
       // Initialize number of parts
       // Since there is no value, there is one less part
       this.numberOfParts = (this.callbackArgument == null) ? 7 : 8;
+    } else if (operation == EnumListenerEvent.AFTER_UPDATE_WITH_GENERATE_CALLBACKS) {
+      if (event.isGenerateCallbacks()) {
+        this.action = UPDATE_ACTION;
+      } else {
+        this.action = UPDATE_ACTION_NO_GENERATE_CALLBACKS;
+      }
+      this.numberOfParts = (this.callbackArgument == null) ? 8 : 9;
     }
```

Comment:
How does event.isGenerateCallbacks() get to be true? I only see it be false in my quick test.

Reply:
The idea behind this new operation (`EnumListenerEvent.AFTER_UPDATE_WITH_GENERATE_CALLBACKS`) is that whenever it is used, the value of `event.isGenerateCallbacks()` will be honored. If `event.isGenerateCallbacks()` is true we will have the legacy behavior and if it is false, we will provoke that the callbacks will not be invoked at the receiving side.
I could have done it differently as follows: when this operation is used, then provoke that the callbacks are not invoked at the receiving side no matter what the value of event.isGenerateCallbacks() is.
Do you think it would be more clear to do change it to this other way?